### ISSUE TITLE
Forbid useless methods

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#697]
# Rule request

## Thesis

We need to forbid methods like:

```python
class Test(SomeBase):
    def some(self, param):
        return super().some(param)
```

The rule is:
1. If method consists of a single call 
2. This call is `super`
3. With the same method: `.some`
4. And same params: `(param)`
5. We raise a violation

We can reuse logic from `UselessLambda` for finding the same params.

## Reasoning

This method does not do anything. It should be removed.
See `useless-super-delegation` http://pylint.pycqa.org/en/latest/technical_reference/features.html

This is OOP rule.

**Comment by bellum** [comment]
@sobolevn I had wanted to mention that is important to note that super() arguments should be Test and self to mark this method as useless but I found W608 violation. Does it mean I should assume super() does not have any arguments always?

**Comment by bellum** [comment]
@sobolevn Also one more thing is that it seems to me that something like this should not be considered useless:

```python
class Test(SomeBase):
    @decorator
    def some(self, param):
        return super().some(param)
```

What do you think?

**Comment by sobolevn** [comment]
@bellum yes, indeed. We discourage usage of arguments to `super()`, but if they are provided, then yes - parameters should match.

Decorators are also important! Thanks for bringing this up. 
Do you want to take this one?

**Comment by bellum** [comment]
@sobolevn Yes, I do. Thanks.

As `wemake_python_styleguide.visitors.ast.functions.FunctionDefinitionVisitor` has already 7 methods I would like to move part related to generator check to `GeneratorDefinitionVisitor` and also probably move wemake_python_styleguide.visitors.ast.functions.FunctionDefinitionVisitor#_get_variable_name to logic/variables.py.

So the changes related to this issue would be done as the part of `FunctionDefinitionVisitor`.

**Comment by sobolevn** [comment]
@bellum 👍
